solc/CommandLineParser.cpp

```diff
@@ -464,6 +464,8 @@ bool CommandLineParser::parseLibraryOption(string const& _input)
 				serr() << "Invalid address for library \"" << libName << "\": " << addrString << endl;
 				return false;
 			}
+			if (libName.find(':') == string::npos)
+				libName = ":" + libName;
```

Comment:
Let's make it an error. User won't be able to specify an empty name for a file containing a library on the command line anyway.

```suggestion
			if (libName.find(':') == string::npos)
			{
				serr() << "Expected a fully qualified library name but got \"" << libName << "\". ";
				serr() << "A fully qualified name must be prepended with the name of the source unit where the library is defined." << endl;
				return false;
			}
```

Reply:
I thought you wanted to treat both the following cases in same way:
- `:Lib`
- `Lib`

Also, after applying your suggestion I should add colon to the `semanticTests`.

Reply:
Well, it's a bit complicated. We want them to be the same in tests because we're already using `L` as a shorthand for `:L` there. But in other places, e.g. in `linkersymbol()` we consider them different. See https://github.com/ethereum/solidity/pull/12073#discussion_r728867808. There I summarized my reasoning and listed all the cases and explained where it's supposed to be different and where it's not.